client/components/Assignments/AssignmentEditor.tsx

```diff
@@ -182,6 +186,14 @@ export class AssignmentEditorComponent extends React.Component {
                 desk: value,
                 filteredUsers: getUsersForDesk(value, this.props.users),
             });
+
+            // Remove the validation error for the "desk" field if PLANNING_AUTO_ASSIGN_TO_WORKFLOW is false
+            if (!appConfig.planning_auto_assign_to_workflow) {
```

Comment:
Directly above this we're calling `this.onChange`, inside this function we're calling `setState` with the new list of errors.

Adding in a `setState` here would mean 2 state updates, it is preferable if we can do this in 1 state update

Reply:
Actually looking further into this, you might not even need any changes here at all.
Instead, what you were to check the config and apply validation here: https://github.com/superdesk/superdesk-planning/blob/develop/client/validators/assignments.tsx#L4

Reply:
Done, Thanks Mark!!